docs/REST API/rest-api.md

```diff
@@ -4301,8 +4301,12 @@ OS vulnerability data is currently available for Windows and macOS. For other pl
     {
       "id": 121,
       "name": "Google Chrome.app",
-      "package_available_for_install": "GoogleChrome.pkg",
-      "self_service": true,
+      "software_package": {
+        "name": "GoogleChrome.pkg",
+        "version": "125.12.0.3",
+        "self_service": true
+      },
+      "app_store_app": null
```

Comment:
Hey @mna, sorry for change again. I had chat with @noahtalerman and we were thinking how could we make Software related API endpoints as consistent as possible. We think it's better to do those changes now since we just released software features and it's easier to break API.

Change: remove `available_for_install` and moving `self_service` flag into `software_package` or `app_store_app` object. This way it will work the same way as `GET /software/titles/:id` endpoint. Available for install is implied if one of the objects is not null (`software_package` or `app_store_app`)

We want same change for device's software `GET /device/:token/software` endpoint. Will follow up with PR change.

Reply:
@marko-lisica @jahzielv just to clarify, because it becomes increasingly hard to keep track of the changes in the PR and across comments and discussions, so to recap, from the current state of the code, my understanding is that we want:

* `GET /software/titles/:id` : get a specific software title: keep it as it currently is, it already has `software_package` and `app_store_app` objects for the specific information of those parts, and it doesn't have any misplaced top-level field AFAICS.
* `GET /hosts/:id/software` : get host's software, remove `available_for_install`, move `self_service` inside `software_package` and `app_store_app`. Those sub-objects were already added as a recent change (with just the `name` and `version` for package, `app_store_id` and `version` for VPP app).
* `GET /device/:token/software`: get my device's software: exactly the same changes as for get host's software.
* `GET /software/titles`: list software titles, this is the biggest change, introduce the `app_store_app` and `software_package` sub-objects, remove `available_for_install`, move `self_service` under the sub-objects, move `icon_url` under `app_store_app`.

Does that capture the requested changes?

Reply:
@mna Exactly, thanks for the summary. I added a link to your comment into the subtask that I created for these changes [here](https://github.com/fleetdm/fleet/issues/20536).

> GET /device/:token/software: get my device's software: exactly the same changes as for get host's software.

For this one we want to have same response as for `GET /hosts/:id/software` as we discussed yesterday.

Reply:
Thanks for the summary @mna ! Thanks for capturing it in a ticket @marko-lisica !

Reply:
@marko-lisica @jahzielv  if we are removing `available_for_install` are we meant to render the available for install icon based on the presence of `software_package` or `app_store_app`? 
![image](https://github.com/user-attachments/assets/61af8e4b-51b4-4163-9800-0abc0add7783)

Reply:
@ghernandez345 that sounds right to me!